components/MapFilters.js

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import {
+  Button,
+  ButtonGroup,
+  Card,
+  CardTitle,
+  Col,
+  Container,
+  Form,
+  FormGroup,
+  Input,
+  Label,
+  Row,
+} from 'reactstrap';
+import Map from '../components/Map';
+import Modal from '../components/Modal';
+import FontAwesome from 'react-fontawesome';
+
+// get most recent month of data
+const getDefaultDates = () => {
+  const date = new Date();
+  let dd = date.getDate();
+  let mm = date.getMonth() + 1;
+  const yyyy = date.getFullYear();
+  if (dd < 10) {
+    dd = '0' + dd;
+  }
+  if (mm < 10) {
+    mm = '0' + mm;
+  }
+  const twoMonth = mm - 2;
+  const threeMonths = mm - 3;
+  const from = yyyy + '-' + '0' + threeMonths + '-' + dd;
+  const to = yyyy + '-' + '0' + twoMonth + '-' + dd;
+  return {
+    fromDate: from,
+    toDate: to,
+  };
+};
+
+export default class MapFilters extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modeSelection: 'any',
+      featuresQuery: '1=1',
+      fromDate: getDefaultDates().fromDate,
+      toDate: getDefaultDates().toDate,
+    };
+    this.filterModes = this.filterModes.bind(this);
+    this.filterFromDate = this.filterFromDate.bind(this);
+    this.filterToDate = this.filterToDate.bind(this);
+  }
+
+  filterModes(modeSelection) {
+    this.setState({ modeSelection });
+  }
+
+  filterFromDate(e) {
+    this.setState({ fromDate: e.target.value });
+  }
+
+  filterToDate(e) {
+    this.setState({ toDate: e.target.value });
+  }
+
+  render() {
+    return (
+      <div id="filterModesDiv">
```

Comment:
Clear out this ID if it’s not being used, and any others that you may come across.

Typically with React the only IDs you need are when using `<label>`s.

Reply:
good to know. Didn’t to a thorough search, but will keep in mind as I update stuff